pkg/util/enforcement_action.go

```diff
@@ -30,14 +49,69 @@ var ErrEnforcementAction = errors.New("unrecognized enforcementAction")
 // spec.enforcementAction field as it was not a string.
 var ErrInvalidSpecEnforcementAction = errors.New("spec.enforcementAction must be a string")
 
-func ValidateEnforcementAction(input EnforcementAction) error {
-	for _, n := range supportedEnforcementActions {
-		if input == n {
-			return nil
+var ErrInvalidSpecScopedEnforcementAction = errors.New("spec.scopedEnforcementAction must be in the format of []{action: string, enforcementPoints: []{name: string}}")
+
+func ValidateEnforcementAction(input EnforcementAction, item map[string]interface{}) error {
+	switch input {
+	case Scoped:
+		return ValidateScopedEnforcementAction(item)
+	case Dryrun, Deny, Warn:
+		return nil
+	default:
+		return fmt.Errorf("%w: %q is not within the supported list %v",
+			ErrEnforcementAction, input, supportedEnforcementActions)
+	}
+}
+
+func ValidateScopedEnforcementAction(item map[string]interface{}) error {
+	obj, err := GetScopedEnforcementAction(item)
+	if err != nil {
+		return fmt.Errorf("error fetching scopedEnforcementActions: %w", err)
+	}
+
+	// validating scopedEnforcementActions
+	for _, scopedEnforcementAction := range *obj {
+		if scopedEnforcementAction.Action == "" {
+			return ErrInvalidSpecEnforcementAction
+		}
+		switch EnforcementAction(scopedEnforcementAction.Action) {
+		case Dryrun, Deny, Warn:
+		default:
+			return fmt.Errorf("%w: %q is not within the supported list %v", ErrEnforcementAction, scopedEnforcementAction.Action, supportedScopedActions)
+		}
+		if len(scopedEnforcementAction.EnforcementPoints) == 0 {
+			return ErrInvalidSpecEnforcementAction
+		}
+		for _, enforcementPoint := range scopedEnforcementAction.EnforcementPoints {
+			if enforcementPoint.Name == "" {
```

Comment:
IMO, we should also return error when a scopedEP does not match any of the supported EPs and persist as an error as part of the constraint's status. otherwise, the bad scopedEP issue will be hard to detect.

Reply:
Agreed. I added the code. If folk do not agree with this I can remove it.